core/src/main/java/org/apache/gravitino/storage/relational/service/ModelVersionMetaService.java

```diff
@@ -0,0 +1,278 @@
+    ModelVersionEntity entityWithNewVersion =
+        ModelVersionEntity.builder()
+            .withVersion(newVersion)
+            .withModelIdentifier(modelIdent)
+            .withAliases(modelVersionEntity.aliases())
+            .withComment(modelVersionEntity.comment())
+            .withUri(modelVersionEntity.uri())
+            .withProperties(modelVersionEntity.properties())
+            .withAuditInfo(modelVersionEntity.auditInfo())
+            .build();
+
+    ModelVersionPO.Builder builder = ModelVersionPO.builder();
+    builder
+        .withModelId(modelPO.getModelId())
+        .withMetalakeId(modelPO.getMetalakeId())
+        .withCatalogId(modelPO.getCatalogId())
+        .withSchemaId(modelPO.getSchemaId());
+    ModelVersionPO modelVersionPO =
+        POConverters.initializeModelVersionPO(entityWithNewVersion, builder);
+
+    List<ModelVersionAliasRelPO> aliasRelPOs =
+        POConverters.initializeModelVersionAliasRelPO(entityWithNewVersion, modelPO.getModelId());
+
+    try {
+      SessionUtils.doMultipleWithCommit(
+          () ->
+              SessionUtils.doWithoutCommit(
+                  ModelVersionMetaMapper.class,
+                  mapper -> mapper.insertModelVersionMeta(modelVersionPO)),
+          () -> {
+            if (aliasRelPOs.isEmpty()) {
+              return;
+            }
+
+            SessionUtils.doWithoutCommit(
+                ModelVersionAliasRelMapper.class,
+                mapper -> mapper.insertModelVersionAliasRels(aliasRelPOs));
+          },
+          () ->
+              // If the model version is inserted successfully, update the model latest version.
+              SessionUtils.doWithoutCommit(
+                  ModelMetaMapper.class,
+                  mapper -> mapper.updateModelLatestVersion(modelPO.getModelId())));
+    } catch (RuntimeException re) {
+      ExceptionUtils.checkSQLException(
+          re, Entity.EntityType.MODEL_VERSION, modelVersionEntity.modelIdentifier().toString());
+      throw re;
+    }
+  }
+
+  public boolean deleteModelVersion(NameIdentifier ident) {
+    NameIdentifierUtil.checkModelVersion(ident);
+
+    NameIdentifier modelIdent = NameIdentifier.of(ident.namespace().levels());
+    // Will throw a NoSuchEntityException if the model does not exist.
+    ModelEntity modelEntity;
+    try {
+      modelEntity = ModelMetaService.getInstance().getModelByIdentifier(modelIdent);
+    } catch (NoSuchEntityException e) {
+      return false;
+    }
+
+    boolean isVersionNumber = NumberUtils.isCreatable(ident.name());
+
+    AtomicInteger modelVersionDeletedCount = new AtomicInteger();
+    SessionUtils.doMultipleWithCommit(
+        // Delete model version relations first
+        () ->
+            modelVersionDeletedCount.set(
+                SessionUtils.doWithoutCommitAndFetchResult(
+                    ModelVersionMetaMapper.class,
+                    mapper -> {
+                      if (isVersionNumber) {
```

Comment:
What if the value of a model alias is all numbers? How can you tell whether it's a model alias or version number?

Reply:
Yes, that's the problem of this solution. One possible solution is to check if the version number exists.

The major limitation is that the current entity store's interface can only accept name identifier as the key to get the object. So we have to set the version number or alias into the name identifier to make it work.

I think the problem does exists, but it is occasionally, we can fix it when we really hit such corner case.

Reply:
Given the situation you mentioned earlier, I think we should add some limitations to the alias name when adding a model version. For example, the name of an alias cannot be all numbers, including formats like `1L`, `2D`, `3.1f`.

Reply:
Yeah, adding a check on the client side is also acceptable.